File: Net2.0/Examples/Access/VB/Example04/Form1.vb
﻿Imports LateBindingApi.Core

Imports Access = NetOffice.AccessApi
Imports NetOffice.AccessApi.Enums
Imports NetOffice.AccessApi.Constants

Imports DAO = NetOffice.DAOApi
Imports NetOffice.DAOApi.Enums
Imports NetOffice.DAOApi.Constants

Public Class Form1

    Private Sub buttonSelectDatabase_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonSelectDatabase.Click

        Dim ofd As New OpenFileDialog()
        ofd.Filter = "(*.mdb)|*.mdb|(*.accdb)|.accdb"

        If (DialogResult.OK = ofd.ShowDialog(Me)) Then

            textBoxFilePath.Text = ofd.FileName
            treeViewInfo.Nodes.Clear()
            ShowDatabaseInfo(textBoxFilePath.Text)

        End If

    End Sub

    Private Sub ShowDatabaseInfo(ByVal filePath As String)

        'Initialize Api COMObject Support
        LateBindingApi.Core.Factory.Initialize()

        ' start access 
        Dim accessApplication As New Access.Application()

        'open database
        Dim database As DAO.Database = accessApplication.DBEngine.Workspaces(0).OpenDatabase(filePath)

        Dim tnTableDefs As TreeNode = treeViewInfo.Nodes.Add("Tables")
        For Each item As DAO.TableDef In database.TableDefs
            tnTableDefs.Nodes.Add(item.Name)
        Next item

        Dim tnQueryDefs As TreeNode = treeViewInfo.Nodes.Add("Queries")
        For Each item As DAO.QueryDef In database.QueryDefs
            tnQueryDefs.Nodes.Add(item.Name)
        Next item

        Dim tnRelations As TreeNode = treeViewInfo.Nodes.Add("Relations")
        For Each item As DAO.Relation In database.Relations
            tnRelations.Nodes.Add(item.Name)
        Next item

        Dim tnContainers As TreeNode = treeViewInfo.Nodes.Add("Containers")
        For Each item As DAO.Container In database.Containers
            tnContainers.Nodes.Add(item.Name)
        Next item

    End Sub

End Class

File: Net2.0/Examples/Access/C#/Example04/Form1.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.IO;
using System.Windows.Forms;

using LateBindingApi.Core;

using Access = NetOffice.AccessApi;
using NetOffice.AccessApi.Enums;
using NetOffice.AccessApi.Constants;

using DAO = NetOffice.DAOApi;
using NetOffice.DAOApi.Enums;
using NetOffice.DAOApi.Constants;

namespace Example04
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void buttonSelectDatabase_Click(object sender, EventArgs e)
        {
            OpenFileDialog ofd = new OpenFileDialog();
            ofd.Filter = "(*.mdb)|*.mdb|(*.accdb)|.accdb";
            if (DialogResult.OK == ofd.ShowDialog(this))
            {
                textBoxFilePath.Text = ofd.FileName;
                treeViewInfo.Nodes.Clear();
                ShowDatabaseInfo(textBoxFilePath.Text);
            }
        }

        private void ShowDatabaseInfo(string filePath)
        {
            // Initialize NetOffice
            LateBindingApi.Core.Factory.Initialize();

            // start access 
            Access.Application accessApplication = new Access.Application();

            // open database
            DAO.Database database = accessApplication.DBEngine.Workspaces[0].OpenDatabase(filePath);

            TreeNode tnTableDefs = treeViewInfo.Nodes.Add("Tables");
            foreach (DAO.TableDef item in database.TableDefs)
                tnTableDefs.Nodes.Add(item.Name); 

            TreeNode tnQueryDefs = treeViewInfo.Nodes.Add("Queries");
            foreach (DAO.QueryDef item in database.QueryDefs)
                tnQueryDefs.Nodes.Add(item.Name); 

            TreeNode tnRelations = treeViewInfo.Nodes.Add("Relations");
            foreach (DAO.Relation item in database.Relations)
                tnRelations.Nodes.Add(item.Name); 
            
            TreeNode tnContainers = treeViewInfo.Nodes.Add("Containers");
            foreach (DAO.Container item in database.Containers)
                tnContainers.Nodes.Add(item.Name); 
        }
    }
}
